### ISSUE TITLE
Individual artists appear twice in the More Like This section in multi folder music library

### ISSUE THREAD

**Issue by guiweber** [jellyfin/jellyfin#12976]
### This issue respects the following points:

- [x] This is a **bug**, not a question or a configuration issue; Please visit our forum or chat rooms first to troubleshoot with volunteers, before creating a report. The links can be found [here](https://jellyfin.org/contact/).
- [x] This issue is **not** already reported on [GitHub](https://github.com/jellyfin/jellyfin/issues?q=is%3Aopen+is%3Aissue) _(I've searched it)_.
- [x] I'm using an up to date version of Jellyfin Server stable, unstable or master; We generally do not support previous older versions. If possible, please update to the latest version before opening an issue.
- [x] I agree to follow Jellyfin's [Code of Conduct](https://jellyfin.org/docs/general/community-standards.html#code-of-conduct).
- [x] This report addresses only a single issue; If you encounter multiple issues, kindly create separate reports for each one.

### Description of the bug

Tested on a fresh install of Jellyfin 10.10.1 on a new system (clean slate).

If an artist's albums are split into two different folder in a music library, the artist sometimes appear twice in the More Like This section.

This may happen, for example, if a user has different folders for hit FLAC and MP3 files, with the same artist having different albums in both. This happens even though each of the library's folders follow the music library guidelines (https://jellyfin.org/docs/general/server/media/music/) and even though artist names are exactly the same in the audio file tags and the artist folder names.

```
/music_flac/artist/album_1
/music_mp3/artist/album_2
```

### Reproduction steps

There seems to be some randomness in how the More Like This section is generated so you may have to browse a bit to notice the issue.

1. Create a folder structure such as the same artist has different albums in two different root folders:
```
/music_1/artist_1/album_1
/music_2/artist_1/album_2
```
2. Add music from other artists to `/music_1` and/or `/music_2`, making sure the genre tag in the audio files matches that of artist_1
3. Create a music library in Jellyfin
4. Add the `/music_1` folder to the music library.
5. Add the `/music_2` folder to the music library.
6. Do a full scan of the library
7. Browse, the artists pages from the Jellyfin UI until you see artist_1 duplicated in the More Like This section.

### What is the current _bug_ behavior?

Individual artists are sometimes duplicated in the More Like This  recommendation section.

### What is the expected _correct_ behavior?

Individual artists should only appear once in the More Like This section

### Jellyfin Server version

10.10.0+

### Specify the build version

10.10.1

### Environment

- OS: TrueNAS
- Virtualization: Docker
- Clients: AndroidTV, Android, Browser
- Browser: Firefox

### Jellyfin logs

None applicable as far as I can tell

### Relevant screenshots or videos

![Image](https://github.com/user-attachments/assets/f3bf2439-cd8c-4c6b-9fed-27e637b15a03)

### Additional information

This may  be linked to https://github.com/jellyfin/jellyfin/issues/11189

**Comment by Dan-613** [comment]
Is this not by design? Are the duplicates showing in your screenshot individual albums?
When I perform your test, my More Like This shows individual albums.
![Image](https://github.com/user-attachments/assets/a9d2d287-b3a9-484c-bbb0-9b1325efaf79)

**Comment by guiweber** [comment]
That's because you're looking at an album page.

If you look at an artist's page, the More Like This section links to other artists, not albums.

**Comment by Dan-613** [comment]
I don't get a More Like This on the artist page.
Are you using a plug-in or toggle it somewhere?

**Comment by guiweber** [comment]
That's strange, I've always seen More Like This on artist pages, even in older version... And it's not something you can enable/disable in settiings as far as I know.

Do you have more artists in your library with albums tagged with the same genre?

I'm using the following standard plugins: Cover Art Archive, Fanart.tv, Reports, Playback Reporting, Bookshelf, OpenSubtitles. I don't think any of those is supposed to provide artist recommendations. But I'll disable everything and restart the server and see if I get the same view.

**Comment by guiweber** [comment]
So I disabled all the plugins and I'm still seeing the issue...

![Image](https://github.com/user-attachments/assets/1dc00b7c-4109-4011-8978-3dd5ff33fe2f)

**Comment by felix920506** [comment]
Please organize your media according to the naming and folder scheme in the docs. 

https://jellyfin.org/docs/general/server/media/music

**Comment by guiweber** [comment]
@felix920506 Hi, this was only an example, the issue also happens when following the guidelines at https://jellyfin.org/docs/general/server/media/music/ if you add multiple folders to your library.

I think the issue should be reopened.

**Comment by guiweber** [comment]
@felix920506 I have updated the report. Let me know if any additional info is needed. Thanks!

**Comment by densilent** [comment]
> All problems mentioned here is due to some design flaws inherited from emby. Although some looks confusing but that is how it defines artists. Those artists scanned from paths requires a physical path related to it and that design could cause all kind of weird behavior when the path is shared between different libraries and/or the path does not exactly have one to one the actual artist. I do see such design very flawed and should not be implemented in the beginning, but unfortunately I don't have a concrete plan on how to migrate from that ATM.

Is this connected to that issue too?
https://github.com/jellyfin/jellyfin/issues/12764

**Comment by gnattu** [comment]
Yes. Artists are global but albums are "inside libraries" but the "physical path in filesystem" of the artist is once again a path under a library folder. Is is designed this bad :(

**Comment by densilent** [comment]
It is unfortunate. This leads not only to "cosmetic" issues in the web interface but to real problems with access rights to different libraries for different users (you are forced to give everyone access to all music libraries).

**Comment by guiweber** [comment]
@felix920506 I don't know if you had time to review the updated reports, but it this is going to stay closed as "not planned" it would be good to update the music guidelines and recommend that each artist be present in one library folder only.

**Comment by densilent** [comment]
>>recommend that each artist be present in one library folder only.
I don't think it is a good recommendation. While it seems OK for a single library, it doesn't work with multiple libraries. The main purpose of many libraries with the same type of content (e.g. Music) is to differentiate content among different users. And different users will inevitably have some artists overlap in their libraries. And that leads to all sorts of problems we discussed here and in my issue too.

Anyway, I think it is a good idea to provide some warning on music guidelines page about this issues.

**Comment by guiweber** [comment]
Still present.

**Comment by farfalleflickan** [comment]
Not stale, rather not fixed...

**Comment by farfalleflickan** [comment]
Still not fixed...

**Comment by guiweber** [comment]
Yup, still present on 10.11.2 after doing full library scan as well as missing metadata scan.

**Comment by Zec-Wicks** [comment]
The issue is still present on 10.11.6

**Comment by theguymadmax** [comment]
Fixed by https://github.com/jellyfin/jellyfin/pull/16807.  Fix will be included in the upcoming 12.0 release.
